game/ato/flightplans/shiprecoverytanker.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Iterator, Type
+from game.ato.flightplans.standard import StandardFlightPlan, StandardLayout
+from game.ato.flightplans.ibuilder import IBuilder
+from game.ato.flightplans.standard import StandardLayout
+from game.ato.flightplans.waypointbuilder import WaypointBuilder
+from game.ato.flightwaypoint import FlightWaypoint
+from game.utils import feet
+
+
+@dataclass(frozen=True)
+class RecoveryTankerLayout(StandardLayout):
+    nav_to: list[FlightWaypoint]
+    recovery_ship: FlightWaypoint
+    nav_from: list[FlightWaypoint]
+
+    def iter_waypoints(self) -> Iterator[FlightWaypoint]:
+        yield self.departure
+        yield from self.nav_to
+        yield self.recovery_ship
+        yield from self.nav_from
+        yield self.arrival
+        if self.divert is not None:
+            yield self.divert
+        yield self.bullseye
+
+
+class RecoveryTankerFlightPlan(StandardFlightPlan[RecoveryTankerLayout]):
+    @staticmethod
+    def builder_type() -> Type[Builder]:
+        return Builder
+
+    @property
+    def tot_waypoint(self) -> FlightWaypoint:
+        return self.layout.recovery_ship
+
+    @property
+    def mission_departure_time(self) -> timedelta:
+        return timedelta(hours=2)
+
+    @property
+    def patrol_start_time(self) -> timedelta:
+        return self.package.time_over_target
+
+    @property
+    def patrol_end_time(self) -> timedelta:
+        return self.tot + self.mission_departure_time
+
+    def tot_for_waypoint(self, waypoint: FlightWaypoint) -> timedelta | None:
+        if waypoint == self.tot_waypoint:
+            return self.tot
+        return None
+
+    def depart_time_for_waypoint(self, waypoint: FlightWaypoint) -> timedelta | None:
+        if waypoint == self.tot_waypoint:
+            return self.tot + self.mission_departure_time
+        return None
+
+
+class Builder(IBuilder[RecoveryTankerFlightPlan, RecoveryTankerLayout]):
+    def layout(self) -> RecoveryTankerLayout:
+
+        # TODO: Propagate the ship position.
+        ship = self.package.target.position
+
+        builder = WaypointBuilder(self.flight, self.coalition)
+
+        recovery = builder.recovery_tanker(ship)
+
+        tanker_type = self.flight.unit_type
+        altitude = tanker_type.preferred_patrol_altitude
+
+        return RecoveryTankerLayout(
+            departure=builder.takeoff(self.flight.departure),
+            nav_to=builder.nav_path(self.flight.departure.position, ship, altitude),
```

Comment:
... oh. It's actually the _cruise_ altitude? If that's all, mind `s/altitude/cruise_altitude/` and leave a comment like "we don't have per-aircraft cruise altitudes recorded yet, so just reuse the patrol altitude"?